From 40899c4c25c5461d7c323d98753e890f1f516192 Mon Sep 17 00:00:00 2001
From: mpvader <mvader@victronenergy.com>
Date: Sun, 6 Jul 2025 22:56:46 +0200
Subject: [PATCH 1/3] package.json: no more optional packages,
 signalk-venus-plugin and more

1) change optional packages to mandatory
This changes nothing to the build, as they were already always installed. But
not anymore as per the recent change to npm-online-install.bbclass.

For the reasons why each package is wanted see here:
https://docs.google.com/document/d/1oJLH7MoRAPoqlo9aaO0gnBYA05a2ruQWFsexZfqWm9E

For why we don't want packages to be listed as optional, see here:
https://github.com/victronenergy/venus/wiki/npm-and-nodejs-packages

Part of https://github.com/victronenergy/venus/issues/1304

2) add signalk-venus-plugin because we need that

3) update @mxtommy/kip and @signalk/freeboard-sk

And do that such that its done automatically when refreshing the shrinkwrap
---
 package.json | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/package.json b/package.json
index 7bf5efd2..b603b973 100644
--- a/package.json
+++ b/package.json
@@ -118,11 +118,9 @@
     "swagger-ui-express": "^4.5.0",
     "unzipper": "^0.10.10",
     "uuid": "^8.1.0",
-    "ws": "^7.0.0"
-  },
-  "optionalDependencies": {
-    "@mxtommy/kip": "^3.2.0",
-    "@signalk/freeboard-sk": "^2.0.0-beta.3",
+    "ws": "^7.0.0",
+    "@mxtommy/kip": "^4.0.0",
+    "@signalk/freeboard-sk": "^2.18.0",
     "@signalk/instrumentpanel": "0.x",
     "@signalk/set-system-time": "^1.2.0",
     "@signalk/signalk-to-nmea0183": "^1.0.0",
@@ -130,8 +128,9 @@
     "@signalk/vesselpositions": "^1.0.0",
     "mdns": "^2.5.1",
     "serialport": "^11.0.0",
-    "signalk-n2kais-to-nmea0183": "^1.3.1",
-    "signalk-to-nmea2000": "^2.16.0"
+    "signalk-n2kais-to-nmea0183": "^2.0.0",
+    "signalk-to-nmea2000": "^2.16.0",
+    "signalk-venus-plugin": "^2.2.0"
   },
   "devDependencies": {
     "@eslint-react/eslint-plugin": "^1.45.1",
-- 
2.34.1

